node/src/types/transaction/transaction_hash_with_approvals.rs

```
@@ -35,9 +36,43 @@ impl TransactionHashWithApprovals {
         transaction_hash: TransactionV1Hash,
         approvals: BTreeSet<TransactionV1Approval>,
     ) -> Self {
-        Self::V1 {
+        Self::V1(TransactionV1HashWithApprovals::new(
             transaction_hash,
             approvals,
+        ))
+    }
+
+    pub(crate) fn new_from_hash_and_approvals(
+        hash: &TransactionHash,
+        approvals: &BTreeSet<TransactionApproval>,
+    ) -> Self {
+        match hash {
+            TransactionHash::Deploy(deploy_hash) => {
+                let approvals: BTreeSet<_> = approvals
+                    .iter()
+                    .filter_map(|approval| match approval {
+                        TransactionApproval::Deploy(approval) => Some(approval.clone()),
+                        TransactionApproval::V1(_) => {
+                            error!("can not add 'transaction' approval to 'legacy deploy'");
```

I think it would be prefferable to return an error in case we get malformed data like this rather than leaking the error into the log.